File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        int slow = n;
        int fast = n;
        do {
            slow = getSquaresSum(slow);
            fast = getSquaresSum(getSquaresSum(fast));
        }while(slow != fast);
        if(slow == 1){
            return true;
        } else {
            return false;
        }
    }
    
    public int getSquaresSum(int n){
        
            int sum = 0;
            while(n > 0){
                int rem = n % 10;
                n = n / 10;
                sum = sum + rem * rem;
            }
            return sum;
    }
}

File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        def getSquaresSum(num):
            sums = 0
            while num > 0:
                rem = num % 10
                sums = sums + (rem * rem)
                num = num // 10
            return sums
            
        slow = n
        fast = n
        while True:
            slow = getSquaresSum(slow)
            fast = getSquaresSum(getSquaresSum(fast))

            if slow == 1 or slow == fast:
                break
        if slow == 1:
            return True
        else:
            return False
            